File: src/dataset/python/sieve_exercism-ThomasZumsteg/claude/ClaudeSonnet4_sieve_v3.py
"""Finds primes numbers"""

def sieve(lim):
    """Implementation of sieve of Eratosthenes"""
    if lim < 2:
        return []
    if lim == 2:
        return [2]
    
    is_prime = [True] * (lim + 1)
    is_prime[0] = is_prime[1] = False
    
    for i in range(2, int(lim**0.5) + 1):
        if is_prime[i]:
            for j in range(i * i, lim + 1, i):
                is_prime[j] = False
    
    return [i for i in range(2, lim + 1) if is_prime[i]]

File: src/dataset/javascript/sieve_exercism-ThomasZumsteg/claude/ClaudeSonnet4_sieve_v3.js
var Sieve = function(size) {
    this.primes = [];
    
    if (size < 2) return;
    
    var sieve = new Array(size + 1);
    sieve.fill(true);
    sieve[0] = sieve[1] = false;
    
    this.primes.push(2);
    
    for (var n = 3; n * n <= size; n += 2) {
        if (sieve[n]) {
            for (var m = n * n; m <= size; m += 2 * n) {
                sieve[m] = false;
            }
        }
    }
    
    for (var i = 3; i <= size; i += 2) {
        if (sieve[i]) {
            this.primes.push(i);
        }
    }
}

module.exports = Sieve;